src/pooling/problem_description.cpp

```diff
@@ -0,0 +1,202 @@
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include <miopen/pooling/problem_description.hpp>
+#include <miopen/mlo_internal.hpp>
+#include <miopen/pooling.hpp>
+
+#include <sstream>
+
+namespace miopen {
+
+namespace pooling {
+
+template <typename T>
+std::string get_vect_config(const std::vector<T>& v)
+{
+    std::string str;
+    for(auto itr = v.begin(); itr < v.end(); itr++)
+    {
+        str += (std::to_string(*itr) + (itr == v.end() - 1 ? "" : "x"));
+    }
+    return str;
+}
+
+NetworkConfig ProblemDescription::MakeNetworkConfig() const
+{
+    std::ostringstream ss;
+
+    int pooling_method =
+        (pooling.GetMode() == miopenPoolingMax)
+            ? MLO_POOLING_OP_MAX
+            : ((pooling.GetMode() == miopenPoolingAverage) ? MLO_POOLING_OP_AVE
+                                                           : MLO_POOLING_OP_AVE_INCLUSIVE);
+
+    ss << "m" + std::to_string(pooling_method);
+
+    if(direction == Direction::Forward)
+    {
+
+        ss << "_i" << static_cast<int>(save_index);
+        ss << "_dt" << xDesc.GetType();
+        ss << "_ker" << get_vect_config(pooling.lens);
+        ss << "_str" << get_vect_config(pooling.strides);
+        ss << "_it" << pooling.GetIndexType();
+
+        if(xDesc.GetSize() == 4)
+        {
+            int batch_sz, n_outputs, out_height, out_width;
+            std::tie(batch_sz, n_outputs, out_height, out_width) =
+                miopen::tien<4>(yDesc.GetLengths(), 1);
+
+            const int _out_pix_tile0 = 1;
+            int _out_pix_tile1       = out_height <= 8 ? 1 : out_height <= 32 ? 4 : 8;
+            if(out_height > 16 && out_height % 32 > 16)
+                _out_pix_tile1 =
+                    std::min(16, std::max(1, prePow2(_out_pix_tile1 * pooling.strides[0])));
+
+            int _grp_tile0 = out_width <= 8 ? 8 : (out_width % 32 <= 16 ? 16 : 32);
+            int _grp_tile1 =
+                out_height <= 8
+                    ? 8
+                    : out_height < 16 ? 16 : out_height <= 32 ? 32 : out_height <= 64 ? 64 : 128;
```

Comment:
@DrizztDoUrden Please check handling of 16. For 8, 32 and 64 we use `<=`, but we use `<` for 16. I am unable to quickly find in this PR where this irregularity came from.

Reply:
This is copy of the original source: https://github.com/ROCmSoftwarePlatform/MIOpen/blob/ca809fee745e5ca18d74607e9b9472ca0232b7c9/src/ocl/mloPooling.cpp#L140
Link is to the commit before the PR.
If it is an error, it is unrelated to the PR and should be investigated by someone familiar with the implementation, kernel and pooling in general.
I agree that this particular place is suspicious. It may be a dirty uncommented hack for some weird behaviour or whatever. Nonetheless, it should be investigated and probably fixed.